consensus/consensus-types/src/proof_of_store.rs

```diff
@@ -321,10 +331,19 @@ impl ProofOfStore {
         }
     }
 
-    pub fn verify(&self, validator: &ValidatorVerifier) -> anyhow::Result<()> {
-        validator
+    pub fn verify(&self, validator: &ValidatorVerifier, cache: &ProofCache) -> anyhow::Result<()> {
+        if let Some(signature) = cache.get(&self.info) {
+            if signature == self.multi_signature {
```

Comment:
this is fine today since there's only leader aggregates the signature, but it's possible that different nodes aggregate different 2f+1